src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java

```diff
@@ -28,552 +16,488 @@
 import org.bukkit.plugin.Plugin;
 import org.bukkit.util.Vector;
 
-import be.seeseemelk.mockbukkit.UnimplementedOperationException;
-import be.seeseemelk.mockbukkit.command.MessageTarget;
-import be.seeseemelk.mockbukkit.metadata.MetadataTable;
+import java.util.*;
+import java.util.concurrent.LinkedTransferQueue;
+
+import static org.junit.Assert.*;
+
+public abstract class EntityMock implements Entity, MessageTarget {
+    private final UUID uuid;
+    private final Queue<String> messages = new LinkedTransferQueue<>();
+    private final Set<PermissionAttachment> permissionAttachments = new HashSet<>();
+    String customName = null;
+    EntityDamageEvent e = null;
```

Comment:
```suggestion
    EntityDamageEvent lastEntityDamageEvent = null;
```
I would say it's better name for that.